sshd-contrib/src/main/java/org/apache/sshd/contrib/server/session/proxyprotocolv2/utils/ProxyUtils.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.sshd.contrib.server.session.proxyprotocolv2.utils;
+
+import java.util.Arrays;
+
+import org.apache.sshd.common.util.buffer.Buffer;
+import org.apache.sshd.common.util.buffer.BufferUtils;
+
+
+/**
+ * Proxy Utilities class
+ *
+ * @author Oodrive - François HERBRETEAU (f.herbreteau@oodrive.com)
+ */
+public final class ProxyUtils {
+
+    private ProxyUtils() {
+        // Utility Class
+    }
+
+    /**
+     * Read an unsigned short value in network order.
+     *
+     * @param  buffer the buffer to read
+     * @return        an unsigned short
+     */
+    public static int readUnsignedShort(Buffer buffer) {
+        // Read the expected bytes from the buffer
+        byte[] datas = new byte[2];
+        buffer.getRawBytes(datas);
+        // Extends the array with 2 more bytes
+        datas = Arrays.copyOf(datas, 4);
+        // Extract the value
+        int value = (int) (BufferUtils.getUInt(datas, 0, Integer.BYTES) >> Short.SIZE);
+        return value & 0xFFFF;
```

Comment:
This is a bit strange. Why not simply `return Buffer.getShort() & 0xFFFF;` ? (Compare commit c411efb83)

Reply:
You're right. This way is much better. I'll fix this.

Reply:
We could add one. Feel free to do so :-) It wasn't needed so far; the SSH protocol doesn't use 16bit values. (I think there is one extension that uses a 16bit value somewhere, check the call hierarchy of getShort() -- which is also why nobody noticed until recently that the implementation was completely broken.)

Reply:
Thx for the info.

I will add a `Buffer.getUShort()` to improve the code readability of my implementation.